### PR TITLE
feat: initial features

### PR DESCRIPTION
**Pull request by xbcsmith**

### CODE REVIEW

File: src/epr/config.py
Comment (caproven) [inline src/epr/config.py]: Should be lists of respective types from `models` to match usage. `main.py` has insertions like
```python
event = Event()
...
cfg.events.append(event)
```

File: src/epr/models.py
Comment (caproven) [inline src/epr/models.py]: By convention should be PascalCase. Also applies to `Event_Receiver_Group` model

File: src/epr/main.py
Comment (caproven) [inline src/epr/main.py]: Fields arg wasn't working for me, found that this (& equivalent for other resource commands) needs `.split(",")`. Otherwise the value remains a string whose characters are later joined in `client.py`'s `_fields = ",".join(fields)`.
```sh
❯ eprcli search event-receiver --id 01HWJZSMRM26ZSCQFKEA2MMHEN --fields 'name'
{'events': [], 'event_receivers': [{'errors': [{'message': 'Cannot query field "n" on type "EventReceiver".', 'locations': [{'line': 1, 'column': 79}]}, {'message': 'Cannot query field "a" on type "EventReceiver".', 'locations': [{'line
': 1, 'column': 81}]}, {'message': 'Cannot query field "m" on type "EventReceiver".', 'locations': [{'line': 1, 'column': 83}]}, {'message': 'Cannot query field "e" on type "EventReceiver".', 'locations': [{'line': 1, 'column': 85}]}]}], 'event_receiver_groups': []}
```

File: src/epr/main.py
Comment (caproven) [inline src/epr/main.py]: Should mention comma separated/delimited. Applies to `--fields` flag for other commands as well

File: src/epr/client.py
Comment (caproven) [inline src/epr/client.py]: Suggest creating dataclass for graph queries so the return here is self documenting, e.g.,
```python
@dataclass
class GraphQLQuery:
    query: str
    variables: str
```
Same could be useful for `new_graphql_mutation_query()` as well.

File: src/epr/client.py
Comment (caproven) [inline src/epr/client.py:20]: Suggest keeping `search_*` & `get_*` methods as-is and making remainder "private" (`__` prefix?) so consumers know what's intended for use.

File: src/epr/config.py
Comment (caproven) [inline src/epr/config.py]: I couldn't tell from usage, why are these lists? Seems like they're only ever length 1 based on cli args

**Review by caproven — COMMENTED** [review]
I'll mention that python tooling drives me insane & I had to hack things to even run the cli, but it did work through some basic scenarios I tried.

Only blatant bug was handling of `--fields` (see comments). Other comments are nice-to-haves, I don't want to hold this initial work up.

File: src/epr/models.py
Comment (xbcsmith) [reply src/epr/models.py]: Fixed

File: src/epr/client.py
Comment (xbcsmith) [reply src/epr/client.py:20]: _ is gentleman's private. To be clear epr-python is intended to be a library + client. The CLI as a convenience/example. While not obvious now there may be uses for the functions later.

File: src/epr/config.py
Comment (xbcsmith) [reply src/epr/config.py]: When using epr-python as a library for a larger program you could easily pass a list of dicts to the the config.

Comment (xbcsmith) [reply src/epr/config.py]: Decided the library should not require the overhead of the Dataclasses.

File: src/epr/main.py
Comment (xbcsmith) [reply src/epr/main.py]: fixed

File: src/epr/client.py
Comment (xbcsmith) [reply src/epr/client.py]: Good suggestion. Based on my intention that Dataclasses should not be required to use the library I might not implement this.

File: src/epr/config.py
Comment (caproven) [reply src/epr/config.py]: If that's the case then please call `.as_dict()` on `main.py`'s lines 518, 528, & 538 (where these lists are appended to).

Comment (caproven) [reply src/epr/config.py]: Maybe missing something. As a library, the epr client in this repo doesn't use the config - it's only for the cli. Perhaps `Client` should take a config but it probably doesn't need this stuff since you'd just use method params

File: src/epr/client.py
Comment (caproven) [reply src/epr/client.py:20]: Methods like `post()`, `query()`, & `mutation()` don't seem meant for use outside the class, I don't know about the rest. I'll leave it to your judgement.

File: src/epr/client.py
Comment (caproven) [reply src/epr/client.py]: I'm fine with this but could you please explain why you don't want dataclasses required for library use? Are you trying to minimize what's exposed in the lib, or is it something with dataclasses specifically?

File: src/epr/config.py
Comment (xbcsmith) [reply src/epr/config.py]: decided you were right

**Review by caproven — APPROVED** [review]

File: src/epr/main.py
Comment (caproven) [inline src/epr/main.py]: What's the significance of the number 26? Contextually I'd guess it's expected ID length, in which case making a const or leaving a comment may help

File: docs/README.md
Comment (caproven) [inline docs/README.md]: Should be `EventReceiverGroup`

File: docs/README.md
Comment (caproven) [inline docs/README.md]: Should be `EventReceiverGroup()`

**Review by caproven — CHANGES_REQUESTED** [review]

**Review by caproven — APPROVED** [review]

**Pull request merged**
